WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailProductListAdapter.kt

```diff
@@ -29,12 +28,15 @@ class OrderDetailProductListAdapter(
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
         val item = orderItems[position]
-        val productId = ProductHelper.productOrVariationId(item.productId, item.variationId)
         val imageSize = holder.view.resources.getDimensionPixelSize(R.dimen.image_minor_100)
-        val productImage = PhotonUtils.getPhotonImageUrl(productImageMap.get(productId), imageSize, imageSize)
+        val productImage = PhotonUtils.getPhotonImageUrl(productImageMap.get(item.productId), imageSize, imageSize)
         holder.view.initView(orderItems[position], productImage, formatCurrencyForDisplay)
         holder.view.setOnClickListener {
-            productItemListener.openOrderProductDetail(productId)
+            if (item.variationId == 0L) {
+                productItemListener.openOrderProductDetail(item.productId)
+            } else {
+                productItemListener.openOrderProductVariationDetail(item.productId, item.variationId)
+            }
```

Comment:
np: We could use the `isVariation` property from `Order.Item` to improve a little bit the readability for this conditional since it's not directly implied that `0L` means a non-variation product. WDYT?
```suggestion
           if (item.isVariation) {
                productItemListener.openOrderProductVariationDetail(item.productId, item.variationId)
            } else {
                productItemListener.openOrderProductDetail(item.productId)
            }
```